ERCS/erc-x.md

```diff
@@ -0,0 +1,166 @@
+---
+eip: 7550
+title: Transient ERC-20 Approvals
+description: Transient ERC-20 approvals via function calls and EIP-712 secp256k1 signatures
+author: Joey Santoro (@joeysantoro), Moody Salem (@moodysalem)
+discussions-to: TODO
+status: Draft
+type: Standards Track
+category: ERC
+created: 2023-10-26
+requires: 20, 712, 1153, 2612
+---
+
+## Abstract
+This standard specifies a time-bound transient [ERC-20](./erc-20.md) token approval mechanism as a cheap augmentation of standard token approval functionality.
+
+It includes a separate transient (single transaction) mapping of `transientAllowance` which can be set via a `transientApprove` call or a `transientPermit` signature.
+
+`transientPermit` signatures allow the approved contract to spend an unlimited amount of tokens until the specified deadline. The approval is *intentionally replayable* in order to save an additional SSTORE and remove the need for nonces.
+
+## Motivation
+
+[ERC-20](./erc-20.md) approvals empower complex smart contract operation of token balances. These are typically achieved either natively or via [ERC-2612](./erc-2612.md) `permit`.
+
+The `allowance` of a token is persistent until revoked or used. This has two drawbacks:
+* gas expensive: requires an SSTORE to set and allowance and a second to use it
+* security risk: if an approved contract is later found to be insecure then approved funds are at risk
+
+EIP-1153 Transient Storage allows a parallel cheap approval mechanism which is bound to a single transaction. This simultaneously addresses both issues of persistent allowances.
+
+## Specification
+
+Compliant contracts must implement 3 new functions + 1 event in addition to ERC-20 and ERC-2612:
+
+```sol
+function transientAllowance(address owner, address spender) external view returns (uint)
+function transientPermit(address owner, address spender, uint deadline, uint8 v, bytes32 r, bytes32 s) external
+function transientApprove(address spender, uint amount) external
+
+event TransientApproval(address indexed _owner, address indexed _spender, uint256 _value)
+```
+
+`transientAllowance`, `transientApprove`, and `TransientApproval` function identically to `allowance`, `approve`, `Approval` respectively, with the caveats that:
+ - `transientAllowance` is uses EIP-1153 transient storage to track transient approvals
+ - tokens SHOULD exclusively exhaust the transient allowance before checking or using the persistent allowance for `transferFrom`.
```

Comment:
You should define "persistent allowance" (presumably as the traditional ERC-20 allowance.)